jre_emul/android/platform/libcore/luni/src/test/java/libcore/java/util/jar/JarFileTest.java

```diff
@@ -0,0 +1,1067 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package libcore.java.util.jar;
+
+//import java.io.ByteArrayOutputStream;
+//import java.io.File;
+//import java.io.FileOutputStream;
+//import java.io.IOException;
+//import java.io.InputStream;
+//import java.net.URL;
+//import java.security.CodeSigner;
+//import java.security.InvalidKeyException;
+//import java.security.InvalidParameterException;
+//import java.security.Permission;
+//import java.security.PrivateKey;
+//import java.security.Provider;
+//import java.security.PublicKey;
+//import java.security.Security;
+//import java.security.SignatureException;
+//import java.security.SignatureSpi;
+//import java.security.cert.Certificate;
+//import java.security.cert.X509Certificate;
+//import java.util.Arrays;
+//import java.util.ArrayList;
+//import java.util.Enumeration;
+//import java.util.List;
+//import java.util.Vector;
+//import java.util.jar.Attributes;
+//import java.util.jar.JarEntry;
+//import java.util.jar.JarFile;
+//import java.util.jar.JarOutputStream;
+//import java.util.jar.Manifest;
+//import java.util.zip.ZipEntry;
+//import java.util.zip.ZipException;
+//import java.util.zip.ZipFile;
+import junit.framework.TestCase;
+//import libcore.io.IoUtils;
+//import tests.support.resource.Support_Resources;
+//
+//
+public class JarFileTest extends TestCase {
+    /**
+     * These test the modified jar feature j2objc doesn't support,
```

Comment:
Move this comment to below the copyright, so the rest of the file can be skipped by future developers.